### PR TITLE
[E2E] macos tray test automation

### PR DESCRIPTION
**Pull request by adrianriobo**
## Solution/Idea

Add E2E automation tests for macos tray. The automation for mac will use native solution based on applescripts to interact with the tray.

This PR includes the implementation for the initial set of e2e scenarios for ux components on MacOS platform. To accommodate that implementation a refactor on the testing structure was required in order to expand and reuse the functionality among all different types of test (integration and e2e) 

Downstream CI run results for this PR:

* [windows](https://crcqe-jenkins-csb-codeready.cloud.paas.psi.redhat.com/blue/organizations/jenkins/qe%2Fbundle_baremetal_windows10-brno/detail/bundle_baremetal_windows10-brno/217/pipeline)
* [mac](https://crcqe-jenkins-csb-codeready.cloud.paas.psi.redhat.com/blue/organizations/jenkins/qe%2Fbundle_baremetal_macos14-brno/detail/bundle_baremetal_macos14-brno/273/pipeline)
* [rhel](https://crcqe-jenkins-csb-codeready.cloud.paas.psi.redhat.com/blue/organizations/jenkins/qe%2Fbundle_baremetal_rhel8-brno/detail/bundle_baremetal_rhel8-brno/174/pipeline)

The required parameters to run the ux scenarios are still missing on the downstream CI, ux feature was run manually on a mac instance with the current CRC version (1.25.0):
```bash
make -C macos_tray e2e GODOG_OPTS="--godog.tags='@darwin && @ux' --godog.format=junit" \
                BUNDLE_LOCATION="--bundle-location=''" \
                PULL_SECRET_FILE="--pull-secret-file=pull-secret" \
                INSTALLER_PATH="--installer-path=crc-macos-amd64.pkg" \
                USER_PASSWORD="--user-password=XXXX"
```
resulting on:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="crc" tests="7" skipped="0" failures="1" errors="0" time="1354.117782704">
  <testsuite name="Basic test" tests="0" skipped="0" failures="0" errors="0" time="0"></testsuite>
  <testsuite name="Behind proxy test" tests="0" skipped="0" failures="0" errors="0" time="0"></testsuite>
  <testsuite name="Certificate rotation test" tests="0" skipped="0" failures="0" errors="0" time="0"></testsuite>
  <testsuite name="End-to-end health check" tests="0" skipped="0" failures="0" errors="0" time="0"></testsuite>
  <testsuite name="Local image to image-registry" tests="0" skipped="0" failures="0" errors="0" time="0"></testsuite>
  <testsuite name="Operator from marketplace" tests="0" skipped="0" failures="0" errors="0" time="0"></testsuite>
  <testsuite name="Test configuration settings" tests="0" skipped="0" failures="0" errors="0" time="0"></testsuite>
  <testsuite name="ux" tests="7" skipped="0" failures="1" errors="0" time="1354.113146961">
    <testcase name="Install CRC" status="passed" time="33.613631749"></testcase>
    <testcase name="Install tray" status="passed" time="75.117393328"></testcase>
    <testcase name="Start Cluster" status="passed" time="476.660750086"></testcase>
    <testcase name="Connect the cluster #1" status="passed" time="16.173251692"></testcase>
    <testcase name="Connect the cluster #2" status="passed" time="30.273120298"></testcase>
    <testcase name="Stop Cluster" status="failed" time="323.139393178">
      <failure message="Step user should get notified with cluster state as stopped: notification: The OpenShift Cluster was successfully stopped. Timeout"></failure>
    </testcase>
    <testcase name="Restart Cluster" status="passed" time="348.496128553"></testcase>
  </testsuite>
</testsuites>
```

*errors are directly realated with [tray-macos issue #84](https://github.com/code-ready/tray-macos/issues/84)

### CODE REVIEW

**Comment by adrianriobo** [comment]
Related to #2190 current install scenario fails randomly

**Comment by guillaumerose** [comment]
Can you rebase ? Otherwise,
/lgtm

Thanks a lot for automating this!

File: test/extended/crc/ux/tray/tray_windows.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+package tray
+
+import (
+	"fmt"
+)
+
+type handler struct {
+	bundleLocation *string
+	pullSecretFile *string
```
Comment (jsliacan) [inline test/extended/crc/ux/tray/tray_windows.go:11]: Are these file contents or is it the file location? `bundleLocation` and `pullSecretLocation` or `bundle` and `pullSecret`. We have these awkwardly inconsistent names in multiple places from the past, so let's try to get rid of them slowly :).

**Review by jsliacan — APPROVED** [review]
Thanks for refactoring :). I left a few minor comments and I didn't check if the build tags are all well placed. But I assume they are since you ran it on all platforms. Otherwise LGTM!

File: test/e2e/features/ux.feature [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Feature: ux
+
+    Tets CRC usage with ux components. 
+        * Handle CRC install/ uninstall operations based on installer distribution
+        * Manage CRC through the tray
+
+    @darwin
+    Scenario: Install CRC 
+        Given a environment where CRC is not installed
+        When install CRC from installer
```
[the remark this answers is not in the document]
Comment (adrianriobo) [reply test/e2e/features/ux.feature:11]: Not sure about this, as to me succeeds is already implicit on the clause. The template inherited is heavy conditioned by the approach used (procedure driven), the intention is trying to get rid of that way to define scenarios and fully embrace behavior driven approach.

File: test/extended/crc/cmd/cmd.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+import (
+	"fmt"
+	"time"
+
+	clicumber "github.com/code-ready/clicumber/testsuite"
+)
+
+const (
+	retryWait                 = "60s"
```
[the remark this answers is not in the document]
Comment (adrianriobo) [reply test/extended/crc/cmd/cmd.go:11]: Change the approach to a timeout based approach.. even with these approach need to pass the expected number of iterations but those are totally dependent on the element we are retrying. so these values should not be common / shared

**Comment by guillaumerose** [comment]
Looks good. Let's use them for the next release.

**Pull request merged**